python/test_infra/ttir_builder.py

```diff
@@ -92,6 +102,17 @@ def generate_and_store_random_golden(self, operand: Operand) -> None:
         golden = Golden(random_tensor, seed)
         self._store_golden(operand, golden)
 
+    def generate_input_golden(self, operand: Operand, index: int) -> None:
+        """
+        Generates random tensor of `input`s shape, assigns it to a golden,
+        and maps `input` to that golden.
+        """
+        seed = self._get_seed()
+        random_tensor = self._generate_random_tensor(self.get_shape(operand), seed)
+        golden = Golden(random_tensor, seed)
+        self.id_golden_map[f"input_{str(index)}"] = golden
+        self._store_golden(operand, golden)
+
```

Comment:
You could expand `generate_and_store_random_golden` with `index: Optional[int] = None` to avoid duplication, if it makes sense to merge them.

If not, you can do

```python
    def generate_and_store_random_golden(self, operand: Operand) -> Golden:
        """
        Generates random tensor of `operand`s shape, assigns it to a golden,
        and maps `operand` to that golden.
        
        Returns generated golden.
        """
        seed = self._get_seed()
        random_tensor = self._generate_random_tensor(self.get_shape(operand), seed)
        golden = Golden(random_tensor, seed)
        self._store_golden(operand, golden)
        
    def generate_input_golden(self, operand: Operand, index: int) -> None:
        """
        Generates random tensor of `input`s shape, assigns it to a golden,
        and maps `input` to that golden.
        """
        self.id_golden_map[f"input_{str(index)}"] = self.generate_and_store_random_golden(operand)
```

Reply:
Btw you don't need `f"input_{str(index)}"`, fstring will take care of int cast. You can just do `f"input_{index}"`

Reply:
fixed